### ISSUE TITLE
Feature request: permissive operation (highlight rather than prohibit words)

### ISSUE THREAD

**Issue by ebpa** [aaron-em/ten-hundred-mode.el#1]
I think an essential next feature would be permissive operation (highlight and retain disallowed words).  This would allow you to go back and revise your writing without necessarily interrupting your flow.  Retaining these words in the buffer really simplifies use of a thesaurus tool like synosaurus.

Great mode!  I'm glad to see someone implemented this for Emacs! :+1:

**Comment by aaron-em** [comment]
This seems like a great idea! Thanks for suggesting it.

I spent some time this weekend putting together a flycheck checker that uses the mode's word list. That went well overall, but there's some hinkiness around syntax tables, and flycheck strongly expects to be used in the context of a major mode, to the extent where I'm not actually sure it is possible for an arbitrary minor mode to define and invoke flycheck checkers without potentially breaking the user's configuration.

The upshot is that the next release of ten-hundred-mode will either define a major, instead of a minor, mode, or it will define both a major and a minor mode and make the permissive behavior easily available only in the context of the major mode. (You'd still be able to `M-x flycheck-mode RET M-x flycheck-select-checker RET ten-hundred-mode RET` with the minor mode active, but that's the point - you'd have to do it by hand, instead of getting it for free, because I can't see a way to activate the checker with certainty that I'm not hosing checkers already active, or in addition to them.)

I'm not sure which would be the preferable option. You're the first user I know I have, so I'd very much like your input here.

**Comment by ebpa** [comment]
"First!" (I'm honored :-)

I'm sure I wasn't the first and that many people havIf you solicit feedback
in your README.org I bet other people will chime in with feature
suggestions.e tried your package already.  New packages are shown at the
top of package.el _Packages_ buffer, the MELPA twitter bot tweets new
packages/updates, and Sacha included your package in her weekly Emacs News
round-up (http://sachachua.com/blog/2016/04/2016-04-11-emacs-news/).  If
you solicit feedback in your README.org I bet other people will chime in
with ideas / feature suggestions.

I'm afraid I'm only a consumer of flycheck, so I can't speak to those
particular design challenges.  emacs.stackexchange.com has been a very
helpful resource for me and will certainly have some useful insight to
share if you ask.

I had only imagined it being customizeable via a variable and had not
thought about minor vs major modes.  Now that you've got me thinking I
suppose there are applications for both.  A major mode would offer basic
text editing and probably be effectively fundamental-mode +
ten-hundred-mode.  I could also imagine use as a minor mode (for example
with org-mode).  Since you mention these alternative possibilities the
notion of a general minor mode sounds particularly awesome.  I'm imagining
writing code and only being able to use reserved words and the "ten
hundred" most common words in variable/class names and comments :-)  Sounds
like an incredibly fun exercise!

**Comment by aaron-em** [comment]
You're right; I should do more to solicit feedback, and will in the next release. Thanks for letting me know about ten-hundred-mode being featured in Sacha's news roundup, that's awesome! I didn't know that was a thing, and I'll definitely be subscribing. 

I have a couple questions out on the Emacs SE about the flycheck weirdness I mentioned. Its author is active there, so I have high hopes of getting useful responses. If flycheck can't do what I need it to for this purpose, I'll rework permissive mode not to use it; it'll just take a little longer. (But I probably won't try to apply it to variable names and the like - just strings and comments. Or at the very least I'll make that optional.)

Thanks again! Watch this space.
